plugins/record-hook/src/RecordHook.ts

```diff
@@ -22,9 +22,10 @@ export const BulkRecordHook = async (
   options: { chunkSize?: number; parallel?: number } = {}
 ) => {
   try {
+    const recordsData = await event.data
     const records = await event.cache.init<Records>(
       'records',
-      async () => (await event.data).records
+      () => recordsData.records
```

Comment:
I think this is actually a problem. If too hooks initiate at the same time they'll race unless the promise for the records data isn't added to the cache.